src/main/java/taxi/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,216 @@
+                + "m.name, m.country "
+                + "FROM car c "
+                + "INNER JOIN manufacturer m "
+                + "ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE c.car_id = ? AND c.deleted = false ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Car car = null;
+            if (resultSet.next()) {
+                car = createCarFromDb(resultSet);
+                car.setDrivers(getDriversForCar(id, connection));
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get car with id = "
+                    + id, ex);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.car_id, c.manufacturer_id, c.model, m.name, m.country "
+                + "FROM car c "
+                + "INNER JOIN manufacturer m "
+                + "ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE c.deleted = false ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                Car currentCar = createCarFromDb(resultSet);
+                currentCar.setDrivers(getDriversForCar(currentCar.getId(), connection));
+                cars.add(currentCar);
+            }
+            return cars;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get all elements "
+                    + "from car DB", ex);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String queryForCars = "UPDATE car "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE car_id = ? AND deleted = false ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                   PreparedStatement statementForCars = connection.prepareStatement(queryForCars)) {
+            statementForCars.setLong(1, car.getManufacturer().getId());
+            statementForCars.setString(2, car.getModel());
+            statementForCars.setLong(3, car.getId());
+            statementForCars.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't update car " + car.getModel(), ex);
+        }
+        deleteObjectInCarDriverById(car.getId());
+        for (int i = 0; i < car.getDrivers().size(); i++) {
+            insertNewDateInCarDriver(car.getDrivers().get(i).getId(), car.getId());
+        }
+        return car;
```

Comment:
let's use one connection for each insert and not create the new one on each iteration